### PR TITLE
[Merged by Bors] - Gateway service

### LINKED ISSUE #2156: Add new API gateway service (for poet broadcast)
This was implemented in https://github.com/spacemeshos/api/pull/118

Blocking on #2133

### PR DESCRIPTION
**Pull request by lrettig**
Note: #2157 should be merged first; will need to rebase onto those changes or possibly resolve conflicts after it's merged

## Motivation
Closes #2156 

## Changes
- Upgrades to a newer version of the API that includes a new, simple GatewayService with a single endpoint to receive a poet proof to be broadcast
- Implements this GatewayService

## Test Plan
- Adds a test for the new service

### CODE REVIEW

File: api/grpcserver/gateway_service.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+// NewGatewayService creates a new grpc service using config data.
+func NewGatewayService(net api.NetworkAPI) *GatewayService {
+	return &GatewayService{
+		Network: net,
+	}
+}
+
+// BroadcastPoet accepts a binary poet message to broadcast to the network
+func (s GatewayService) BroadcastPoet(_ context.Context, in *pb.BroadcastPoetRequest) (*pb.BroadcastPoetResponse, error) {
+	log.Info("GRPC GatewayService.BroadcastPoet")
```
Comment (antonlerner) [inline api/grpcserver/gateway_service.go:35]: don't we use a custom logger for the GRPC? if not I think we should

File: api/grpcserver/mesh_service.go [hunk 2]
```diff
@@ -357,15 +361,27 @@ func (s MeshService) readLayer(layer *types.Layer, layerStatus pb.Layer_LayerSta
...
 	log.Info("GRPC MeshService.LayersQuery")
 
+	var startLayer, endLayer types.LayerID
+	if in.StartLayer == nil {
+		startLayer = 0
+	} else {
+		startLayer = types.LayerID(in.StartLayer.Number)
+	}
+	if in.EndLayer == nil {
+		endLayer = 0
```
Comment (antonlerner) [inline api/grpcserver/mesh_service.go]: what does this mean? no values will be returned?

File: api/grpcserver/gateway_service.go (hunk 1 above)
Comment (lrettig) [reply api/grpcserver/gateway_service.go:35]: I agree, it's a work in progress, see #2086

File: api/grpcserver/mesh_service.go (hunk 2 above)
Comment (lrettig) [reply api/grpcserver/mesh_service.go]: yup

**Review by antonlerner — APPROVED** [review]

**Comment by lrettig** [comment]
bors merge

**Pull request closed without merging**